src/Raven.Server/Documents/Subscriptions/SubscriptionProcessor/RevisionsSubscriptionProcessor.cs

```diff
@@ -152,11 +153,19 @@ protected override bool ShouldSend((Document Previous, Document Current) item, o
                 var match = Patch.MatchCriteria(Run, DocsContext, transformResult, ProjectionMetadataModifier.Instance, ref result.Data);
                 if (match == false)
                 {
+                    transformResult.Dispose();
+                    result.Data?.Dispose();
                     result.Data = null;
                     reason = $"{item.Current.Id} filtered by criteria";
                     return false;
                 }
 
+                if (transformResult.Location != result.Data.Location)
```

Comment:
What is `Location`? Can you elaborate more on this check?

Reply:
This is the pointer to the start of the blittable.
The check is necessary to determine if the patch allocated a new object into result.Data (which is passed by reference to the Patch script). 
This can occur when a subscription query contains select {}, which triggers the patch to create a new object.

Reply:
ok